cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/AbstractSparkContext.java

```diff
@@ -83,7 +86,7 @@ protected AbstractSparkContext(ApplicationSpecification applicationSpecification
                                  Map<String, String> runtimeArguments, DiscoveryServiceClient discoveryServiceClient,
                                  MetricsContext metricsContext, LoggingContext loggingContext,
                                  @Nullable PluginInstantiator pluginInstantiator,
-                                 @Nullable WorkflowToken workflowToken) {
+                                 @Nullable WorkflowToken workflowToken, @Nullable List<URI> localResources) {
```

Comment:
Also, since only `ClientSparkContext` supports localization, the `AbstractSparkContext` shouldn't have anything to do with it.